lib/rspec/page-regression/matcher.rb

```diff
@@ -9,10 +9,15 @@ module RSpec::PageRegression
 
       @responsive_filepaths = FilePaths.responsive_file_paths(RSpec.current_example, args)
 
-      opt = args.select { |k,_| RENDER_ARGS.include?(k) }
-
-      Renderer.render_responsive(page, @responsive_filepaths, opt)
-      @comparisons = @responsive_filepaths.map{ |filepaths| ImageComparison.new(filepaths) }
+      render_options = args.select { |k,_| RENDER_ARGS.include?(k) }
+      compare_options = {}
+      compare_options[:threshold] = args.fetch(:threshold, RSpec::PageRegression.threshold)
+      compare_options[:mode] = args.fetch(:mode, RSpec::PageRegression.mode)
+      compare_options[:include_rect] = args.fetch(:include_area, nil)
+      compare_options[:exclude_rect] = args.fetch(:exclude_rect, nil)
```

Comment:
@palkan @ronen Currently there's a check to only allow use of `ALLOWED_ARGS` (see https://github.com/teachbase/rspec-page-regression/blob/2f6f9fc4147c9d788ac3bf34d45c19a7050d3a24/lib/rspec/page-regression/matcher.rb#L54). That means users can in any case only use imatcher options which have been whitelisted explicitly. I'd suggest to keep a comprehensive list of all supported imatcher args in `IMATCHER_ARGS` and to merge the values of the used imatcher args into the main hash of `compare_options` (instead of nesting some args into an `imatcher` hash and placing other args (such as `threshold`) at top level).

For details also see https://github.com/rprt/rspec-page-regression/pull/31/files/a4e386fcd0924c91c29cc67ca346e5cf6a90250b#r100282420 
What do you think?